numpy/_core/numeric.py

```diff
@@ -2385,6 +2387,19 @@ def _array_equal_dispatcher(a1, a2, equal_nan=None):
     return (a1, a2)
 
 
+_no_nan_types = {
+    nt.bool,
+    nt.int8,
+    nt.int16,
+    nt.int32,
+    nt.int64,
```

Comment:
```suggestion
    # should use np.dtype.BoolDType, but as of writing
    # that fails the reloading test.
    type(np.dtype(nt.bool)),
    type(np.dtype(nt.int8)),
    type(np.dtype(nt.int16)),
    type(np.dtype(nt.int32)),
    type(np.dtype(nt.int64)),
```

Reply:
that fails as well with `numpy has not attribute dtype`

Reply:
bleh, yah... but it seems explicitly imported already, so no problem...